packages/edit-post/src/index.js

```diff
@@ -20,49 +20,6 @@ import './plugins';
 import Editor from './editor';
 import { store as editPostStore } from './store';
 
-/**
- * Reinitializes the editor after the user chooses to reboot the editor after
- * an unhandled error occurs, replacing previously mounted editor element using
- * an initial state from prior to the crash.
- *
- * @param {Object}  postType     Post type of the post to edit.
- * @param {Object}  postId       ID of the post to edit.
- * @param {Element} target       DOM node in which editor is rendered.
- * @param {?Object} settings     Editor settings object.
- * @param {Object}  initialEdits Programmatic edits to apply initially, to be
- *                               considered as non-user-initiated (bypass for
- *                               unsaved changes prompt).
- */
-export function reinitializeEditor(
```

Comment:
I think we can't just remove a public API function; we have to deprecate it first. 

Once deprecated, we could consider removing it, although this is also debatable - this PR contains a relevant discussion about removing deprecated APIs: 

https://github.com/WordPress/gutenberg/pull/46105

My personal preference is that [we remove deprecated functionality eventually](https://github.com/WordPress/gutenberg/pull/46105#issuecomment-1328880894), but there isn't a general agreement about that.

In any case, we should consider deprecating first.